### PR TITLE
Rename 'options' parameters. Add config snapshots API. Add valid values for SCMType. 

### PR DESCRIPTION
**Pull request by naveedaz**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] Validation errors from the [Linter extension for VS Code](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/linter.md) have all been fixed for this spec. (**Note:** for large, previously checked in specs, there will likely be many errors shown. Please contact our team so we can set a timeframe for fixing these errors if your PR is not going to address them).
  - [x] The spec follows the patterns described in the [Swagger good patterns](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-good-patterns.md) document unless the service API makes this impossible.

### CODE REVIEW

File: arm-web/2016-08-01/WebApps.json [hunk 1]
```diff
@@ -1736,6 +1736,140 @@
...
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Web/sites/{name}/config/web/snapshots/{snapshotId}/recover": {
+      "post": {
+        "tags": [
+          "WebApps"
+        ],
+        "summary": "Gets a snapshot of the configuration of an app, such as platform version and bitness, default documents, virtual applications, Always On, etc.",
```
Comment (veronicagg) [inline arm-web/2016-08-01/WebApps.json]: summary looks inaccurate for this operation, please review

File: arm-web/2016-08-01/WebApps.json [hunk 2]
```diff
@@ -1736,6 +1736,140 @@
...
+        "description": "Gets a snapshot of the configuration of an app, such as platform version and bitness, default documents, virtual applications, Always On, etc.",
```
Comment (veronicagg) [inline arm-web/2016-08-01/WebApps.json]: description looks inaccurate for this operation

Comment (naveedaz) [inline arm-web/2016-08-01/WebApps.json]: @Nking92 Please take a look.

File: arm-web/2016-08-01/WebApps.json (hunk 1 above)
Comment (naveedaz) [inline arm-web/2016-08-01/WebApps.json]: @Nking92 Please take a look.

File: arm-web/2016-08-01/WebApps.json (hunk 2 above)
Comment (Nking92) [inline arm-web/2016-08-01/WebApps.json]: I think the description is accurate, but maybe it's a little confusing. You do get an entire site configuration back. I can change the first part of the sentence to "Gets the configuration of an app at a previous point in time" if that's more clear

File: arm-web/2016-08-01/WebApps.json (hunk 1 above)
Comment (Nking92) [inline arm-web/2016-08-01/WebApps.json]: I'll change this one to "Reverts the configuration of an app to a previous snapshot"

**Review by veronicagg — CHANGES_REQUESTED** [review]

File: arm-web/2016-08-01/WebApps.json [hunk 3]
```diff
@@ -3415,9 +3549,9 @@
             "type": "string"
           },
           {
-            "name": "options",
+            "name": "migrationOptions",
```
Comment (naveedaz) [inline arm-web/2016-08-01/WebApps.json]: This API is fairly new and does not exist in SDKs yet.

**Review by veronicagg — APPROVED** [review]
If you're happy with the latest descriptions for the APIs, rest looks good to me.

**Pull request merged**